src/renderer/components/dialogs/index.js

```diff
@@ -23,5 +23,62 @@ module.exports = {
   ImexProgress,
   About,
   Settings,
-  ForwardMessage
+  ForwardMessage,
+  EncrInfo
+]
+
+class Controller extends React.Component {
+  constructor (props) {
+    super(props)
+
+    var dialogs = {}
+    allDialogs.forEach((Component) => {
+      dialogs[Component.name] = {
+        Component,
+        props: false
+      }
+    })
+
+    this.state = { dialogs }
+    this.close = this.close.bind(this)
+  }
+
+  open (name, props) {
+    var Component = this.state.dialogs[name]
+    if (!Component) throw new Error(`Component with name ${name} does not exist`)
+    if (!props) props = {}
+    this.state.dialogs[name].props = props
+    this.setState({ dialogs: this.state.dialogs })
+  }
+
+  close (name) {
+    this.state.dialogs[name].props = false
+    this.setState({ dialogs: this.state.dialogs })
+  }
+
+  render () {
+    const { saved, userFeedback } = this.props
+    const { dialogs } = this.state
+
+    return (
+      <div>
+        {Object.values(dialogs).map((dialog) => {
+          var name = dialog.Component.name
```

Comment:
perhaps we shouldn't be rendering all dialogs all the time, or we could find some other way to only render when absolutely necessary (when state.dialogs has changed?)